spec/gl-matrix/mat4-spec.js

```diff
@@ -645,6 +645,50 @@ function buildMat4Tests() {
             });
         });
 
+        describe("removeTranslation", function() {
+          beforeEach(function() {
+              let q = quat.create();
+              q = quat.setAxisAngle(q, [0, 1, 0], 0.7);
+              let t = vec3.fromValues(1, 2, 3);
+              let s = vec3.fromValues(5, 6, 7);
+              mat4.fromRotationTranslationScale(matA, q, t, s);
+              mat4.removeTranslation(matB, matA);
+              result = vec3.fromValues(1, 2, 3);
+              mat4.getTranslation(result, matB);
+          })
+          it("should return the null translation", function() { expect(result).toBeEqualish([0, 0, 0]); });
+        });
+
+        describe("removeScaling", function() {
+          beforeEach(function() {
+              let q = quat.create();
+              q = quat.setAxisAngle(q, [0, 1, 0], 0.7);
+              let t = vec3.fromValues(1, 2, 3);
+              let s = vec3.fromValues(5, 6, 7);
+              mat4.fromRotationTranslationScale(matA, q, t, s);
+              mat4.removeScaling(matB, matA);
```

Comment:
`result = mat4.removeScaling(out, matA);` would be better, since that allows you to check more things.
- You can check if it returns the `result`
- If it sets `out` to the correct value
- If matA is untouched